src/composables/staking/useStaking.ts

```diff
@@ -10,50 +10,172 @@ import GaugeFactoryABI from '@/lib/abi/GaugeFactory.json';
+        }
+      },
+      liquidityGauges: {
+        __args: {
+          where: {
+            poolId_in: userPoolsAddresses.value
+          }
+        },
+        poolId: true
+      },
+      liquidityGauge: {
+        __aliasFor: 'liquidityGauges',
+        __args: {
+          where: {
+            poolAddress: poolAddress || ''
+          }
+        },
+        id: true,
+        shares: {
+          __args: {
+            where: {
+              user: account.value.toLowerCase()
+            }
+          },
+          balance: true
+        }
+      }
+    })
+  );
+
+  /**
+   * COMPUTED
+   * Need to wrap the extracted query response vars into
+   * computed properties so they retain reactivity
+   * when returned by this composable
+   */
+  const userGaugeShares = computed(() => {
+    if (!gaugeSharesRes.value?.gaugeShares) return [];
+    return gaugeSharesRes.value.gaugeShares;
+  });
+
+  const userLiquidityGauges = computed(() => {
+    if (!gaugeSharesRes.value?.liquidityGauges) return [];
+    return gaugeSharesRes.value.liquidityGauges;
+  });
+
+  const stakedShares = computed(() => {
+    if (!poolAddress || isFetchingStakingData.value) return '0';
+
+    // the first share object in the list will be the share for our user
+    // as we filtered via pool address and user in the query
+    const stakedShares =
+      gaugeSharesRes.value?.liquidityGauge[0]?.shares[0]?.balance;
+    return stakedShares;
+  });
 
-  function getStakeState(pool: DecoratedPoolWithStakedShares) {
-    // just random logic for differentiating between stake states // TODO REPLACE
-    if (pool.stakedPct === '1') {
-      return StakeState.MaxStaked;
-    } else {
-      return StakeState.CanStake;
+  /**
+   * METHODS
+   */
+  async function stakeBPT() {
+    if (!poolAddress) {
+      throw new Error(
+        `Attempted to call stake, however useStaking was initialised without a pool address.`
+      );
     }
+    const gaugeAddress = await getGaugeAddress(poolAddress);
+    const gauge = new LiquidityGauge(gaugeAddress, getProvider());
+    const tx = await gauge.stake(parseUnits(balanceFor(poolAddress), 18));
+    return tx;
   }
 
-  async function stakeBPT() {
-    if (!pool) throw new Error(`Pool not loaded.`);
-    const gaugeAddress = await getGaugeAddress();
+  async function unstakeBPT() {
+    if (!poolAddress) {
+      throw new Error(
+        `Attempted to call unstake, however useStaking was initialised without a pool address.`
+      );
+    }
+    const gaugeAddress = await getGaugeAddress(poolAddress);
     const gauge = new LiquidityGauge(gaugeAddress, getProvider());
-    const tx = await gauge.stake(parseUnits(balanceFor(pool.address), 18));
+    const stakedShares = await getStakedShares();
+    const tx = await gauge.unstake(parseUnits(stakedShares, 18));
     return tx;
   }
 
-  async function getGaugeAddress(): Promise<string> {
-    if (!pool) throw new Error(`Pool not loaded.`);
+  async function getStakedShares() {
```

Comment:
Receiving `stakedShares` from contract pending response from Fabio regarding whether `balance` from gauge subgraph is cumulative or not.